lib/backup/Backup.ts

```diff
@@ -0,0 +1,77 @@
+import os from 'os';
+import fs from 'fs';
+import path from 'path';
+import Config from '../Config';
+import Logger, { Context } from '../Logger';
+import LndClient from '../lndclient/LndClient';
+
+class Backup {
+  private logger!: Logger;
+
+  private backupDir!: string;
+
+  private config = new Config();
+
+  public start = async (args: { [argName: string]: any }) => {
+    await this.config.load(args);
+
+    this.backupDir = this.resolveBackupDir(args.backupdir);
+
+    if (!fs.existsSync(this.backupDir)) {
+      fs.mkdirSync(this.backupDir);
+    }
+
+    this.logger = new Logger({
+      context: Context.Backup,
+      level: this.config.loglevel,
+      filename: this.config.logpath,
+      instanceId: this.config.instanceid,
+      dateFormat: this.config.logdateformat,
+    });
+
+    for (const currency in this.config.lnd) {
+      const lndConfig = this.config.lnd[currency]!;
+
+      if (!lndConfig.disable) {
+        const lndClient = new LndClient(
+          lndConfig,
+          currency,
+          Logger.DISABLED_LOGGER,
+        );
+
+        await lndClient.init();
+        this.listenToChannelBackups(lndClient);
+
+        lndClient.subscribeChannelBackups();
+        this.logger.debug(`Listening to ${currency} LND channel backups`);
+      }
+    }
+
+    this.logger.info('Started backup daemon');
+  }
+
+  private listenToChannelBackups = (lndClient: LndClient) => {
+    lndClient.on('channelBackup', (channelBackup) => {
+      this.logger.verbose(`${lndClient.currency} channel backup: ${channelBackup}`);
+
+      fs.writeFileSync(
```

Comment:
I doubt performance will really be an issue with this tool but still you can use the promisified version of the `fs` API (there are examples elsewhere in the code) and `await` it. Also it would probably be good to check to handle errors in case we can't write to the file here, maybe try writing to a separate file with the date & time in the filename? That might be useful in case we ever get near simultaneous backups and file write attempts.

Reply:
The arrow function of the callback isn't `async` so the events should be handled sequentially and parallel writes shouldn't be a problem (apart from that it is really unlikely with the current behavior of LND that something like this would happen).

I agree on the error handling and will implement that.

Reply:
Sounds good.